tendrl/commons/tests/test_init.py

```diff
@@ -0,0 +1,159 @@
+import py.test
+import maps
+import importlib
+import inspect
+from tendrl.commons import objects
+from tendrl.commons import TendrlNS
+
+#__init__() testing
+def test_constructor():
+
+    '''
+    Testing for constructor involves checking if all needed variales are declared/
+    initialized
+    '''
+
+    tendrlNS = TendrlNS()
+
+    # Default Parameter Testing
+    assert tendrlNS.ns_name == "tendrl"
+    assert tendrlNS.ns_src == "tendrl.commons"
+    #Check for existance and right data type
+    assert isinstance(NS,maps.NamedDict)
+
+
+#Testing _list_modules_in_package_path
+def test_list_modules_in_package_path():
+
+    tendrlNS = TendrlNS("tendrl","tendrl.commons")
+    modules = [('cluster_node_context', 'tendrl.commons.objects.cluster_node_context'), ('cluster_tendrl_context', 'tendrl.commons.objects.cluster_tendrl_context'),\
+              ('cpu', 'tendrl.commons.objects.cpu'),('definition', 'tendrl.commons.objects.definition'), ('detected_cluster', 'tendrl.commons.objects.detected_cluster'),\
+              ('disk', 'tendrl.commons.objects.disk'), ('file','tendrl.commons.objects.file'), ('job','tendrl.commons.objects.job'),\
+              ('memory', 'tendrl.commons.objects.memory'), ('node', 'tendrl.commons.objects.node'), ('node_context', 'tendrl.commons.objects.node_context'),\
+              ('node_network', 'tendrl.commons.objects.node_network'), ('os', 'tendrl.commons.objects.os'), ('platform', 'tendrl.commons.objects.platform'),\
+              ('service', 'tendrl.commons.objects.service'), ('tendrl_context', 'tendrl.commons.objects.tendrl_context')]
+    ns_objects_path = "/home/risjain/Tendrl/commons-develop/tendrl/commons/objects"
+    ns_objects_prefix = "tendrl.commons.objects."
+    ret = tendrlNS._list_modules_in_package_path(ns_objects_path,ns_objects_prefix)
+    #TO-DISCUSS : modules is hard coded and might change in future
+    assert ret == modules
+
+    ret = tendrlNS._list_modules_in_package_path("test","test")
+    assert len(ret) == 0
+
+
+#Testing _register_subclasses_to_ns
+def test_register_subclasses_to_ns():
+
+    tendrlNS = TendrlNS()
+    tendrlNS._register_subclasses_to_ns()
+    #assert getattr(NS,)
+
+
+#Testing _add_object
+def test_add_object():
+    tendrlNS = TendrlNS()
+    obj_name = "test_obj"
+    obj = importlib.import_module("tendrl.commons.objects.cluster_node_context")
+
+    current_ns = tendrlNS._get_ns()
+    for obj_cls in inspect.getmembers(obj, inspect.isclass):
+        tendrlNS._add_object(obj_name,obj_cls[1])
+        break
+
+    assert isinstance(getattr(current_ns.objects,"_test_obj")['atoms'],maps.NamedDict)
+    assert isinstance(getattr(current_ns.objects,"_test_obj")['flows'],maps.NamedDict)
+
+
+#Testing _get_objects
+def test_get_objects():
+    objects_list = ['Service', 'Platform', 'Memory', 'ClusterTendrlContext', 'Node', 'NodeContext',\
```

Comment:
I have made the changes according to what I understood which is creating a directory hierarchy(not including files as it would just be duplication) and using that hierarchy to generate dummy testing data for the functions like get_objects(), I am not sure but were you instructing me to create a fixture where I would create a complete NS (populating it with objects and flows) and then using that NS?